test/Microsoft.Azure.Devices.Edge.Test/EndToEnd.cs

```diff
@@ -0,0 +1,160 @@
+
+            string name = "temp sensor";
+            Log.Information("Running test '{Name}'", name);
+            await Profiler.Run(
+                async () =>
+                {
+                    var iotHub = new IotHub(
+                        Context.Current.ConnectionString,
+                        Context.Current.EventHubEndpoint,
+                        proxy);
+
+                    EdgeDevice device = (await EdgeDevice.GetIdentityAsync(
+                        Context.Current.DeviceId,
+                        iotHub,
+                        token)).Expect(() => new Exception("Device should have already been created in setup fixture"));
+
+                    var builder = new EdgeConfigBuilder(device.Id);
+                    Context.Current.Registry.ForEach(
+                        r => builder.AddRegistryCredentials(r.address, r.username, r.password));
+                    builder.AddEdgeAgent(edgeAgent).WithProxy(proxy, Protocol.Amqp);
+                    builder.AddEdgeHub(edgeHub).WithProxy(proxy, Protocol.Amqp);
+                    builder.AddModule("tempSensor", tempSensor);
+                    await builder.Build().DeployAsync(iotHub, token);
+
+                    var hub = new EdgeModule("edgeHub", device.Id, iotHub);
+                    var sensor = new EdgeModule("tempSensor", device.Id, iotHub);
+                    await EdgeModule.WaitForStatusAsync(
+                        new[] { hub, sensor },
+                        EdgeModuleStatus.Running,
+                        token);
+                    await sensor.WaitForEventsReceivedAsync(token);
+
+                    var sensorTwin = new ModuleTwin(sensor.Id, device.Id, iotHub);
+                    await sensorTwin.UpdateDesiredPropertiesAsync(
+                        new
+                        {
+                            properties = new
+                            {
+                                desired = new
+                                {
+                                    SendData = true,
+                                    SendInterval = 10
+                                }
+                            }
+                        },
+                        token);
+                    await sensorTwin.WaitForReportedPropertyUpdatesAsync(
+                        new
+                        {
+                            properties = new
+                            {
+                                reported = new
+                                {
+                                    SendData = true,
+                                    SendInterval = 10
+                                }
+                            }
+                        },
+                        token);
+                },
+                "Completed test '{Name}'",
+                name);
+        }
+
+        [Test]
+        public async Task ModuleToModuleDirectMethod(
+            [Values("Amqp", "Mqtt")] string edgeToCloud,
```

Comment:
In linux, we also run direct method test using amqpws and mqttws for both upstream protocol and client transport type.

Reply:
We get coverage of the WebSocket-based protocols for the _upstream_ connection when we do proxy testing, so we're good there.

But I'm realizing that I'm setting `UpstreamProtocol` on the other modules too (method sender/receiver, temp sensor) and they don't use that variable--they use `ClientTransportType`. So I need to fix that. I can add AmqpWs and MqttWs for `ClientTransportType` too...

Reply:
Fixed.